networkx/algorithms/traversal/breadth_first_search.py

```diff
@@ -41,24 +45,39 @@ def bfs_edges(G, source, reverse=False):
     Based on http://www.ics.uci.edu/~eppstein/PADS/BFS.py
     by D. Eppstein, July 2004.
     """
+    if depth_limit is not None and depth_limit < 0:
+        return
     if reverse and isinstance(G, nx.DiGraph):
         neighbors = G.predecessors
     else:
         neighbors = G.neighbors
     visited = set([source])
     queue = deque([(source, neighbors(source))])
+    time_to_depth_increase = len(queue)
+    current_depth = 1
+    pending_depth_increase = False
     while queue:
+        if depth_limit is not None and current_depth > depth_limit:
```

Comment:
This is probably simpler if you incorporate it into the `while` condition above:

    while queue and current_depth <= depth_limit

In order to avoid the `depth_limit is not None` check here and above, just add

    if depth_limit is None:
        depth_limit = float('inf')

at the beginning of the function to set the depth limit to infinity.